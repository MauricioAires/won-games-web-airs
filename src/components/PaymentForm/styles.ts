import styled, { css } from 'styled-components'
import { tint } from 'polished'
import media from 'styled-media-query'

import * as ButtonStyles from 'components/Button/styles'

export const Wrapper = styled.div``

export const Body = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.small};
    background: ${theme.colors.white};
  `}
`

export const Footer = styled.div`
  ${({ theme }) => css`
    background: ${tint(-0.2, theme.colors.lightGray)};
    color: ${theme.colors.black};
    font-weight: ${theme.font.bold};
    padding: ${theme.spacings.small};
    display: flex;
    align-items: center;

    ${ButtonStyles.Wrapper} {
      padding-left: ${theme.spacings.xxsmall};
      padding-right: ${theme.spacings.xxsmall};
      outline: 0;
    }

    ${media.lessThan('medium')`
      flex-direction: column;

      ${ButtonStyles.Wrapper} {
        margin-bottom: ${theme.spacings.xxsmall};
      }
    `}
  `}
`
