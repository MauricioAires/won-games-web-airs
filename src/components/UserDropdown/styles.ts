import styled, { css } from 'styled-components'

export const Nav = styled.nav`
  display: flex;
  flex-direction: column;
  width: 26rem;
`

export const Username = styled.span`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.xxsmall};
  `}
`

export const Link = styled.a`
  ${({ theme }) => css`
    cursor: pointer;
    display: flex;
    align-items: center;
    text-decoration: none;
    background: ${theme.colors.white};
    color: ${theme.colors.black};
    padding: ${theme.spacings.xsmall} ${theme.spacings.small};
    transition: background, color, ${theme.transition.default};

    &:not(:last-child) {
      border-bottom: 0.1rem solid ${theme.colors.lightGray};
    }

    &:hover {
      background: ${theme.colors.primary};
      color: ${theme.colors.white};
    }

    > svg {
      width: 2.4rem;
      height: 2.4rem;
    }

    > span {
      margin-left: ${theme.spacings.small};
    }
  `}
`
