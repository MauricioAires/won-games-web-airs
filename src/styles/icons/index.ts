import { AddShoppingCart as AddShoppingCartIcon } from '@styled-icons/material-outlined/AddShoppingCart'
import { RemoveShoppingCart as RemoveShoppingCartIcon } from '@styled-icons/material-outlined/RemoveShoppingCart'
import { Menu2 as MenuIcon } from '@styled-icons/remix-fill/Menu2'
import { ShoppingCart as ShoppingCartIcon } from '@styled-icons/material-outlined/ShoppingCart'
import { Search as SearchIcon } from '@styled-icons/material-outlined/Search'
import { Close as CloseIcon } from '@styled-icons/material-outlined/Close'
import { FavoriteBorder as FavoriteBorderIcon } from 'styled-icons/material'
import { Favorite as FavoriteIcon } from 'styled-icons/material'
import { ArrowBackIos as ArrowBackIosIcon } from '@styled-icons/material-outlined/ArrowBackIos'
import { ArrowForwardIos as ArrowForwardIosIcon } from '@styled-icons/material-outlined/ArrowForwardIos'
import { Email as EmailIcon } from '@styled-icons/material-outlined/Email'
import { Lock as LockIcon } from '@styled-icons/material-outlined/Lock'
import { AccountCircle as AccountCircleIcon } from '@styled-icons/material-outlined/AccountCircle'
import { ErrorOutline as ErrorOutlineIcon } from '@styled-icons/material-outlined/ErrorOutline'
import { Windows as WindowsIcon } from '@styled-icons/fa-brands/Windows'
import { Linux as LinuxIcon } from '@styled-icons/fa-brands/Linux'
import { Apple as AppleIcon } from '@styled-icons/fa-brands/Apple'
import { Download as DownloadIcon } from '@styled-icons/boxicons-solid/Download'
import { Add as AddIcon } from 'styled-icons/material'
import { CreditCard as CreditCardIcon } from 'styled-icons/material'
import { FormatListBulleted as FormatListBulletedIcon } from 'styled-icons/material'
import { ExitToApp as ExitToAppIcon } from 'styled-icons/material'
import { FilterList as FilterListIcon } from 'styled-icons/material'
import { KeyboardArrowDown as KeyboardArrowDownIcon } from '@styled-icons/material-outlined/KeyboardArrowDown'
import { ChevronDown as ChevronDownIcon } from 'styled-icons/bootstrap'
import { CheckCircleOutline as CheckCircleOutlineIcon } from '@styled-icons/material-outlined/CheckCircleOutline'
import { ErrorCircle as ErrorCircleIcon } from '@styled-icons/fluentui-system-filled/ErrorCircle'
import { Done as DoneIcon } from '@styled-icons/material-outlined/Done'

export {
  DoneIcon,
  ErrorCircleIcon,
  CheckCircleOutlineIcon,
  MenuIcon,
  ShoppingCartIcon,
  AddShoppingCartIcon,
  FavoriteBorderIcon,
  SearchIcon,
  FavoriteIcon,
  CloseIcon,
  ArrowBackIosIcon,
  ArrowForwardIosIcon,
  EmailIcon,
  LockIcon,
  AccountCircleIcon,
  WindowsIcon,
  LinuxIcon,
  AppleIcon,
  DownloadIcon,
  AddIcon,
  CreditCardIcon,
  FormatListBulletedIcon,
  ExitToAppIcon,
  FilterListIcon,
  KeyboardArrowDownIcon,
  ChevronDownIcon,
  RemoveShoppingCartIcon,
  ErrorOutlineIcon
}
