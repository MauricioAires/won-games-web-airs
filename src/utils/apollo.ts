import { ApolloClient, HttpLink, NormalizedCacheObject } from '@apollo/client'
import { setContext } from '@apollo/client/link/context'
import { useMemo } from 'react'
import apolloCache from './apolloCache'
import { Session } from 'next-auth/core/types'

let apolloClient: ApolloClient<NormalizedCacheObject | null>

function createApolloCliente(session?: Session | null) {
  const httpLink = new HttpLink({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`
  })

  const authLink = setContext((_, { headers }) => {
    console.log('configuração do apooppp', session)
    const authorization = session?.jwt ? `Bearer ${session.jwt}` : ''

    return {
      headers: {
        ...headers,
        authorization
      }
    }
  })
  // grapQL provider
  return new ApolloClient({
    ssrMode: typeof window === 'undefined', // true
    link: authLink.concat(httpLink),
    cache: apolloCache
  })
}

export function initializeApollo(
  initialState = null,
  session?: Session | null
) {
  /**
   * Modelo de coneção SINGLETON
   */
  // server para verificar se já existe uma instância, para não criar outra
  const apoloClientGlobal = apolloClient ?? createApolloCliente(session)

  // recuperando os dados do cahce
  if (initialState) {
    apoloClientGlobal.cache.restore(initialState)
  }

  // sempre iniciando no  SSR com o cahce limpo
  if (typeof window === 'undefined') {
    return apoloClientGlobal
  }

  apolloClient = apolloClient ?? apoloClientGlobal

  return apolloClient
}

export function useApollo(initialState = null, session?: Session | null) {
  const store = useMemo(
    () => initializeApollo(initialState, session),
    [initialState, session]
  )

  return store
}
