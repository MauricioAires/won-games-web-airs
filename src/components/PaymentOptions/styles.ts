import styled, { css, DefaultTheme } from 'styled-components'
import { tint } from 'polished'
import media from 'styled-media-query'

import * as ButtonStyles from 'components/Button/styles'

export const Wrapper = styled.div``

export const Body = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.small};
    background: ${theme.colors.white};
  `}
`

export const Footer = styled.div`
  ${({ theme }) => css`
    background: ${tint(-0.2, theme.colors.lightGray)};
    color: ${theme.colors.black};
    font-weight: ${theme.font.bold};
    padding: ${theme.spacings.small};
    display: flex;
    align-items: center;

    ${ButtonStyles.Wrapper} {
      padding-left: ${theme.spacings.xxsmall};
      padding-right: ${theme.spacings.xxsmall};
      outline: 0;
    }

    ${media.lessThan('medium')`
      flex-direction: column;

      ${ButtonStyles.Wrapper} {
        margin-bottom: ${theme.spacings.xxsmall};
      }
    `}
  `}
`

export const CardsList = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`

export const CardInfo = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    img {
      margin-right: ${theme.spacings.xxsmall};
    }
  `}
`
const ItemStyles = (theme: DefaultTheme) => css`
  background: ${theme.colors.lightGray};
  border-radius: 0.2rem;
  color: ${theme.colors.black};
  padding: 0 ${theme.spacings.xxsmall};
  height: 5rem;
  cursor: pointer;
  display: flex;
  align-items: center;
`

export const CardItem = styled.label`
  ${({ theme }) => css`
    ${ItemStyles(theme)};

    justify-content: space-between;

    &:not(:last-child) {
      margin-bottom: ${theme.spacings.xxsmall};
    }
  `}
`

export const AddCard = styled.div`
  ${({ theme }) => css`
    ${ItemStyles(theme)};

    svg {
      margin-left: ${theme.spacings.xxsmall};
      margin-right: ${theme.spacings.xsmall};
      width: 2.4rem;
    }
  `}
`
