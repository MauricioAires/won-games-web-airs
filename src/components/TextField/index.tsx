import { useState, InputHTMLAttributes } from 'react'

import * as S from './styles'

export type TextFieldProps = {
  onInputChange?: (value: string) => void
  label?: string
  error?: string
  initialValue?: string
  icon?: React.ReactNode
  iconPosition?: 'rigth' | 'left'
  disabled?: boolean
} & InputHTMLAttributes<HTMLInputElement>

const TextField = ({
  label,
  initialValue = '',
  icon,
  name,
  iconPosition = 'left',
  disabled = false,
  error,
  onInputChange,
  ...props
}: TextFieldProps) => {
  const [value, setValue] = useState(initialValue)

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = e.currentTarget.value
    setValue(newValue)

    !!onInputChange && onInputChange(newValue)
  }

  return (
    <S.Wrapper error={!!error} disabled={disabled}>
      {!!label && <S.Label htmlFor={name}>{label}</S.Label>}
      <S.InputWrapper>
        {!!icon && <S.Icon iconPosition={iconPosition}> {icon} </S.Icon>}
        <S.Input
          type="text"
          iconPosition={iconPosition}
          onChange={onChange}
          value={value}
          disabled={disabled}
          name={name}
          {...(label ? { id: name } : {})}
          {...props}
        />
      </S.InputWrapper>
      {!!error && <S.Error>{error}</S.Error>}
    </S.Wrapper>
  )
}

export default TextField
