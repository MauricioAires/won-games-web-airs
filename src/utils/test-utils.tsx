import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import {
  CartContext,
  CartContextData,
  CartContextDefaultValue
} from 'hooks/use-cart'

import { ThemeProvider } from 'styled-components'
import { light } from 'styles/themes'

export type CustomRenderProps = {
  cartProviderProps?: CartContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  {
    cartProviderProps = CartContextDefaultValue,
    ...renderOptions
  }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={light}>
      <CartContext.Provider value={cartProviderProps}>
        {ui}
      </CartContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
