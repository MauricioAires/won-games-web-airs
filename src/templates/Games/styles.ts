import styled, { css } from 'styled-components'
import { Container } from 'components/Container'

import media from 'styled-media-query'

export const Main = styled(Container)`
  ${({ theme }) => css`
    ${media.greaterThan('medium')`
      display: grid;
      grid-template-columns: 26rem 1fr;
      gap: ${theme.grid.gutter};
    `}
  `}
`

export const ShowMore = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 10rem;
`

export const ShowMoreButton = styled.div`
  ${({ theme }) => css`
    text-align: center;
    text-transform: uppercase;
    font-weight: ${theme.font.bold};
    cursor: pointer;
    color: ${theme.colors.white};

    > svg {
      color: ${theme.colors.primary};
    }
  `}
`
