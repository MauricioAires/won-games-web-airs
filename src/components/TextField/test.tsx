import { screen, waitFor, render } from 'utils/test-utils'
import userEvent from '@testing-library/user-event'
import { ShoppingCartIcon } from 'styles/icons'

import TextField from '.'

describe('<TextField />', () => {
  it('Renders with Label', () => {
    render(<TextField label="Label" name="Label" />)

    expect(screen.getByLabelText('Label')).toBeInTheDocument()
  })

  it('Renders without Label', () => {
    render(<TextField />)

    expect(screen.queryByLabelText('Label')).not.toBeInTheDocument()
  })

  it('Renders with placeholder', () => {
    render(<TextField placeholder="hey you" />)

    expect(screen.getByPlaceholderText('hey you')).toBeInTheDocument()
  })

  it('Changes its value when typing', async () => {
    const onInputChange = jest.fn()
    render(
      <TextField
        onInputChange={onInputChange}
        label="TextField"
        name="TextField"
      />
    )

    const input = screen.getByRole('textbox')
    const text = 'This is my new text'

    await userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveValue(text)
      expect(onInputChange).toHaveBeenCalledTimes(text.length)
    })
    expect(onInputChange).toHaveBeenCalledWith(text)
  })

  it('Is accessible by tab', async () => {
    render(<TextField label="TextField" name="TextField" />)

    const input = screen.getByLabelText('TextField')
    expect(document.body).toHaveFocus()

    await userEvent.tab()

    await waitFor(() => {
      expect(input).toHaveFocus()
    })
  })

  it('should render with Icon', () => {
    render(<TextField icon={<ShoppingCartIcon data-testid="icon" />} />)

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render with icon on the rigth side', () => {
    render(
      <TextField
        icon={<ShoppingCartIcon data-testid="icon" />}
        iconPosition="rigth"
      />
    )

    expect(screen.getByTestId('icon').parentElement).toHaveStyle({
      order: 1
    })
  })

  it('does not changes its value when disavled', async () => {
    const onInputChange = jest.fn()

    render(
      <TextField
        onInputChange={onInputChange}
        label="TextField"
        name="TextField"
        disabled
      />
    )

    const input = screen.getByRole('textbox')
    expect(input).toBeDisabled()

    const text = 'This is my new text'

    await userEvent.type(input, text)

    await waitFor(() => {
      expect(input).not.toHaveValue(text)
    })

    expect(onInputChange).not.toBeCalled()
  })

  it('should render with error', () => {
    const { container } = render(
      <TextField
        icon={<ShoppingCartIcon data-testid="icon" />}
        label="TextField"
        error="Erro message"
      />
    )

    expect(screen.getByText(/erro message/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
