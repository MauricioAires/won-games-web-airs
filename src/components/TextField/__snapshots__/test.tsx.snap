// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render with error 1`] = `
.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c4:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c6 {
  width: 2.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #8F8F8F;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c6 > svg {
  width: 100%;
}

.c8 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c8:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c2 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
}

.c0 .c1,
.c0 .c5 {
  color: #FF6347;
}

.c0 .c3 {
  border-color: #FF6347;
}

.c9 {
  color: #FF6347;
  font-size: 1.2rem;
}

<div
  class="c0"
>
  <label
    class="c1 c2"
  >
    TextField
  </label>
  <div
    class="c3 c4"
  >
    <div
      class="c5 c6"
    >
       
      <svg
        aria-hidden="true"
        class="c7"
        data-testid="icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
       
    </div>
    <input
      class="c8"
      type="text"
      value=""
    />
  </div>
  <span
    class="c9"
  >
    Erro message
  </span>
</div>
`
